api/envoy/config/route/v3/route_components.proto

```diff
@@ -760,6 +760,28 @@ message RouteAction {
     ConnectConfig connect_config = 3;
   }
 
+  message MaxStreamDuration {
+    // Specifies the maximum duration allowed for streams on the route. If not specified, the
+    // value from :ref:`max_stream_duration
+    // <envoy_api_field_config.core.v3.HttpProtocolOptions.max_stream_duration>` is used.
+    google.protobuf.Duration max_stream_duration = 1;
+
+    // If present, and the request contains a `grpc-timeout` header
+    // <https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md>`_, use that value or its
+    // default value (infinity) as the *max_stream_duration*, but limit the applied timeout to the
```

Comment:
If we're applying this only when the `grpc-timeout` header is present, then there's no default to consider here.

Reply:
Very true.  Removed.